engine/filesync/localfs.go

```diff
@@ -196,6 +199,45 @@ func (local *localFS) Sync( //nolint:gocyclo
 	var hardlinks []*hardlinkChange
 	var hardlinkMu sync.Mutex
 
+	// Track spans per top-level directory
+	type rootPathSpan struct {
+		span             trace.Span
+		start            time.Time
+		writtenBytes     int64
+		maxWriteDuration time.Duration
+	}
+	rootPathSpans := make(map[string]*rootPathSpan)
+	var rootPathSpansMu sync.Mutex
+
+	getRootPath := func(path string) *rootPathSpan {
+		rootPath := strings.Split(path, string(os.PathSeparator))[0]
```

Comment:
I don't think `os.PathSeparator` will work as intended here cause it will always be linux-based since this code runs in the engine.

Reply:
right.. I assume for windows users this path might arrive as `\\` or `/`, correct? https://cs.opensource.google/go/go/+/master:src/internal/filepathlite/path_windows.go;l=19-20

I assume the  best way to proceed here is to always do the `\\` | `/` check? Mostly checking if by any chance you know we're doing a similar handling somwhere else.